tests/acceptance/00_basics/01_compiler/CFE-2698.cf

```diff
@@ -0,0 +1,153 @@
+body common control
+{
+    inputs => { "../../default.cf.sub" };
+    bundlesequence => { default($(this.promise_filename)) };
+}
+
+bundle agent test
+{
+  meta:
+
+      "description" -> { "CFE-2698" }
+        string => "Test that the behavior of unless matches if => not()";
+
+  vars:
+
+      # Normal case - skip promise based on condition:
+      "skipped_variable_if_not"
+        string => "value",
+        if => not("any");
```

Comment:
*variable*? I don't see a variable in this promise. Just nitpicking the name. I understand that !any evaluates false, so the promise will be skipped.

Reply:
oh skipped variable, .... vars promise i understand.